sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -460,6 +460,40 @@ abstract class SchemaPruningSuite
     checkAnswer(query4, Row(2, null) :: Row(2, 4) :: Nil)
   }
 
+  testSchemaPruning("select nested field in window function") {
+    val windowSql =
+      """
+        |with contact_rank as (
+        |  select row_number() over (partition by address order by id desc) as __rank,
+        |  contacts.*
+        |  from contacts
+        |)
+        |select name.first , __rank from contact_rank
+        |where name.first = 'Jane' AND __rank = 1
```

Comment:
the same comment here, too: https://github.com/apache/spark/pull/28898/files#r446499910